go-controller/cmd/ovn-k8s-cni-overlay/app/helper_windows.go

```diff
@@ -209,31 +235,210 @@ func PlatformSpecificCleanup(args *skel.CmdArgs) error {
+	fakeAddr, fakeAddrNet, _ := net.ParseCIDR("1.2.3.4/32")
+	fakeResult := &current.Result{
+		IPs: []*current.IPConfig{
+			{
+				Version:   "4",
+				Interface: current.Int(1),
+				Address:   net.IPNet{IP: fakeAddr, Mask: fakeAddrNet.Mask},
+				Gateway:   net.ParseIP("1.2.3.4"),
+			},
+		},
+	}
+	return fakeResult
+}
+
 // InitialPlatformCheck checks to see if the container is an infra container
 // by looking at the namespace prefix. If it's not, then it has nothing to
-// do and the CNI should stop here.
-func InitialPlatformCheck(args *skel.CmdArgs) (bool, *current.Result) {
+// do and the CNI should stop here for Windows Server version greated than windowsVersionRTM.
+// For Windows Server version windowsVersionRTM it will configure the pod interface.
+func InitialPlatformCheck(args *skel.CmdArgs, argsMap map[string]string) (bool, *current.Result, error) {
+	windowsVersion, err := getWindowsVersion()
+	if err == nil && windowsVersion <= windowsVersionRTM {
+		namespace := argsMap["K8S_POD_NAMESPACE"]
+		podName := argsMap["K8S_POD_NAME"]
+		ifaceID := fmt.Sprintf("%s_%s", namespace, podName)
+		if strings.Compare(args.Netns, noNameNetNS) != 0 && !checkContainerOVSPort(ifaceID) {
+			result, err := configureInterfaceRTM(args, argsMap)
+			return true, result, err
+		} else {
+			logrus.Infof("InitialPlatformCheck: CNI has nothing to do (called for infra container or network has been setup or container not running")
+			return true, getFakeCNIResult(), nil
+		}
+
+	}
 	if strings.HasPrefix(args.Netns, containerNetNS) || strings.Compare(args.Netns, noNameNetNS) != 0 {
 		// If it is a normal container from the pod, there is nothing to do.
 		// Also, if it is not an infra container, nothing to do in this case as well.
 		logrus.Infof("InitialPlatformCheck: CNI called for normal container or infra container, nothing to do")
 		// This result is ignored anyway by Kubernetes.
-		fakeAddr, fakeAddrNet, _ := net.ParseCIDR("1.2.3.4/32")
-		fakeResult := &current.Result{
-			IPs: []*current.IPConfig{
-				{
-					Version:   "4",
-					Interface: current.Int(1),
-					Address:   net.IPNet{IP: fakeAddr, Mask: fakeAddrNet.Mask},
-					Gateway:   net.ParseIP("1.2.3.4"),
-				},
-			},
+		return true, getFakeCNIResult(), nil
+	}
+	return false, &current.Result{}, nil
+}
+
+func checkContainerOVSPort(ifaceID string) bool {
+	ovsArgs := []string{
+		"find", "interface", fmt.Sprintf("external_ids:iface-id=%s", ifaceID),
+	}
+	outputFind, err := exec.Command("ovs-vsctl", ovsArgs...).CombinedOutput()
+	if err == nil && len(outputFind) != 0 {
+		return true
+	} else {
+		return false
+	}
+}
+
+// Retrieves the Windows build number using wmic
+func getWindowsVersion() (int, error) {
+	args := []string{"os", "get", "BuildNumber", "/value"}
+	out, err := exec.Command("wmic", args...).Output()
+	if err != nil {
+		return 0, err
+	}
+	buildInfo := strings.Split(string(strings.Trim(string(out), "\" \n\r")), "=")
+	windowsVersion, _ := strconv.Atoi(buildInfo[1])
```

Comment:
We should not ignore the 'error' returned by strconv.Atoi, I think.

Reply:
I don't think the build number can be something else than an integer, but I will add the error handling anyway.